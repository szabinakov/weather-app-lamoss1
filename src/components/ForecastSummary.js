import React from "react";
import "../styles/App.css";
import "../styles/ForecastSummary.css";
import iconData from "../data/iconData.json";

function ForecastSummary(props) {
  const { date, description, icon, temperature, onSelect } = props;
  const formattedDate = new Date(date).toDateString();
  const weatherCode = icon;
  // `${icon.slice(0, 1)}00`;
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{date}</div>
      {formattedDate}
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <img src={iconData[weatherCode]} alt={`${icon}`} />
      </div>
      <div className="forecast-summary__temperature">{temperature.max}</div>
      <div className="forecast-summary__description">{description}</div>
      <button type="button" onClick={() => onSelect(date)}>
        More details
      </button>
    </div>
  );
}

export default ForecastSummary;
