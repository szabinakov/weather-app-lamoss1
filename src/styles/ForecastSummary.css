.forecast-summary {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #fbfbfb;
  padding: 20px;
  margin-right: 20px;
  text-align: center;
  border-radius: 50%;
  border: 2px solid white;
  height: em(50px);
  margin: 0 auto 1rem;
  position: relative;
  width: em(50px);
  position: relative;
  width: 250px;
  height: 200px;
  background: #fff;
  border: 5px solid #000;
  border-radius: 50%;
  overflow: scroll;
  box-shadow: inset -10px 10px 0 10px #ccc;
}
.forecast-summary:hover {
  background-color: #e2001a;
  color: #fff;
  cursor: pointer;
  background-size: cover;
  border-radius: 50%;
}

.forecast-date {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 10px;
}

.forecast-icon {
  margin-bottom: 10px;
  font-size: 60px;
}

.forecast-temperature {
  font-size: 24px;
  font-weight: bold;
  margin-top: 10px;
  color: #e2001a;
}

.forecast-description {
  font-size: 20px;
  margin-top: 10px;
  color: #666;
}
.location-details {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: #fbfbfb;
  border-radius: 10px;
  padding: 20px;
  margin-top: 20px;
}
