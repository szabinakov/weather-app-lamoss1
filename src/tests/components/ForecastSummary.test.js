import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "stub description",
    icon: "200",
    temperature: {
      min: 12,
      max: "22°C",
    },
    onSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getByText("1111111")).toHaveClass("forecast-summary__date");
    expect(getByText("stub description")).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByText("22°C")).toHaveClass("forecast-summary__temperature");
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
  });
});
