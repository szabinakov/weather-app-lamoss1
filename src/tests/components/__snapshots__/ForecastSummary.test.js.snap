// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      1111111
    </div>
    Thu Jan 01 1970
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="200"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      stub description
    </div>
    <button
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
